source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOBaseActionElementPrivate.h

```diff
@@ -0,0 +1,17 @@
+//
+//  ACOBaseActionElementPrivate
+//  ACOBaseActionElementPrivate.h
+//
+//  Copyright © 2017 Microsoft. All rights reserved.
+//
+#import "ACOBaseActionElement.h"
+#import "BaseActionElement.h"
+
+using namespace AdaptiveCards;
+
+@interface ACOBaseActionElement()
+
+- (std::shared_ptr<BaseActionElement> const &)getElem;
```

Comment:
> & [](start = 43, length = 2)

This should return by value rather than by reference. If you return by reference, the shared_ptr becomes invalid if the ACOBaseActionElement is destroyed.

Reply:
Thank you!

---
In reply to: [146323478](https://github.com/Microsoft/AdaptiveCards/pull/799#discussion_r146323478) [](ancestors = 146323478)